### PR TITLE
add editors breakdown

### PR DESCRIPTION
**Pull request by giotab**
In order to see Editors usage as well, I changed the existing LanguagesBreakdown component to be more generic so that both Languages and Editors require less effort to maintain.

github copilot editors pie chart

Here are the most significant changes:

New Features:
* [`src/components/BreakdownsComponent.vue`](diffhunk://#diff-336ba41c9c56951419f144974747eef1815df4bc041e893acd928e0cca3a6d5bR1-R243): Introduced a new component that provides a generic breakdown view. This component can display breakdowns by different keys such as language or editor. It includes a pie chart to visualize the top 5 items by accepted prompts and acceptance rate, and a data table for a detailed breakdown.

Enhancements:
* [`src/components/MainComponent.vue`](diffhunk://#diff-e9eea4ec690e7649e2f812dfa7620e6200d892450dceb379d53e72f9be2894afL31-R32): Replaced the `LanguagesBreakdown` component with the new `BreakdownsComponent`. Added 'editors' to the list of tab items to provide a breakdown by editors. [[1]](diffhunk://#diff-e9eea4ec690e7649e2f812dfa7620e6200d892450dceb379d53e72f9be2894afL31-R32) [[2]](diffhunk://#diff-e9eea4ec690e7649e2f812dfa7620e6200d892450dceb379d53e72f9be2894afL50-R59) [[3]](diffhunk://#diff-e9eea4ec690e7649e2f812dfa7620e6200d892450dceb379d53e72f9be2894afL80-R81)
* [`src/components/MetricsViewer.vue`](diffhunk://#diff-3232f8c11e48e57eba48531f39e477604432e91bafd81119783d472a38ad3f10L161-L163): Removed the processing of language breakdowns as it is now handled by the `BreakdownsComponent`. [[1]](diffhunk://#diff-3232f8c11e48e57eba48531f39e477604432e91bafd81119783d472a38ad3f10L161-L163) [[2]](diffhunk://#diff-3232f8c11e48e57eba48531f39e477604432e91bafd81119783d472a38ad3f10L294-R293)

Refactoring:
* [`src/model/Breakdown.ts`](diffhunk://#diff-e3e6203de0c9c60337f8b245031acb618472570b637017da01d84a7cdc2ccef5L1-R11): Renamed from `Language.ts` and updated the class name to `Breakdown` to make it more generic.
* [`src/model/Metrics.ts`](diffhunk://#diff-935a8c63d4425b72ea64be2c76b0a582c02a950d65bcb628257d182f6fb73babL1-R1): Renamed the `Breakdown` class to `BreakdownData` and updated the `breakdown` member of the `Metrics` class to use `BreakdownData`. [[1]](diffhunk://#diff-935a8c63d4425b72ea64be2c76b0a582c02a950d65bcb628257d182f6fb73babL1-R1) [[2]](diffhunk://#diff-935a8c63d4425b72ea64be2c76b0a582c02a950d65bcb628257d182f6fb73babL37-R37) [[3]](diffhunk://#diff-935a8c63d4425b72ea64be2c76b0a582c02a950d65bcb628257d182f6fb73babL49-R49)

Removals:
* [`src/components/LanguagesBreakdown.vue`](diffhunk://#diff-847369674e1670136d0a40efa29e0812f138aaa6d779c2d92ccb58380ba87020L1-L236): Removed this component as its functionality is now covered by the `BreakdownsComponent`.
* [`src/components/MetricsViewer.vue`](diffhunk://#diff-3232f8c11e48e57eba48531f39e477604432e91bafd81119783d472a38ad3f10L81): Removed the import of the `Language` model as it is no longer used.

### CODE REVIEW

File: src/components/BreakdownsComponent.vue [hunk 1]
```diff
@@ -0,0 +1,243 @@
+<template>
```
Comment (martedesco) [inline src/components/BreakdownsComponent.vue]: Would it make sense to name it BreakdownComponent.vue? As I would assume it is a singular breakdown per component e.g. languages or editors

File: src/components/BreakdownsComponent.vue [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+  <div>
+    <div class="tiles-container">
+      <!-- Acceptance Rate Tile -->  
+      <v-card elevation="4" color="white" variant="elevated" class="mx-auto my-3" style="width: 300px; height: 175px;">
+          <v-card-item>
+            <div>
+              <div class="text-overline mb-1" style="visibility: hidden;">filler</div>
+              <div class="text-h6 mb-1">Number of {{ breakdownKey }}s</div>
```
Comment (martedesco) [inline src/components/BreakdownsComponent.vue]:
```suggestion 
<div class="text-h6 mb-1">Number of {{ breakdownKey }}</div>

Comment (martedesco) [reply src/components/BreakdownsComponent.vue]: Is there a specific constraint to use the keys on the singular? If we use Editors and Languages as key we wouldn't need to add 's' in every time we reference this prop.

File: src/components/MetricsViewer.vue [hunk 3]
```diff
@@ -78,7 +78,6 @@
 <script lang="ts">
 import { defineComponent, ref, toRef } from 'vue';
 import { Metrics } from '../model/Metrics';
-import { Language } from '../model/Language';
```
Comment (martedesco) [inline src/components/MetricsViewer.vue:81]: Thanks for the clean up 👍🏼

**Comment by giotab** [comment]
> @giotab , thanks for your contribution! I added a few comments 👍🏼

Hi @martedesco , not sure I can see the comments 🤔  unless you mean in the PR's description?

**Comment by martedesco** [comment]
> Hi @martedesco , not sure I can see the comments 🤔 unless you mean in the PR's description?

My bad, I didn't submit them!

 > Also, should I bump the version number to 1.5.0?
 
I will add a few things before releasing it, so no need to worry about that 🙏🏼

File: src/components/BreakdownsComponent.vue (hunk 2 above)
Comment (giotab) [reply src/components/BreakdownsComponent.vue]: The breakdownKey is used in this line 
`const breakdownName = breakdownData[props.breakdownKey as keyof typeof breakdownData] as string;` 
in order to get the value, e.g. breakdownName would be `dockerfile` or `vscode`.
```json
{
    "language": "dockerfile",
    "editor": "vscode",
    "suggestions_count": 3,
    "acceptances_count": 0,
    "lines_suggested": 3,
    "lines_accepted": 0,
    "active_users": 1
}
```
(replaces the previous `const languageName = breakdown.language;`)

If we use `Editors`, then the above access will not work.

I updated the code a little bit, to centralise the "pluralisation" of the key.

**Review by martedesco — APPROVED** [review]

**Pull request merged**